assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useState, useMemo, useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { v4 as uuidv4 } from 'uuid';
+import { trackEvent } from '@web-stories-wp/tracking';
+
+/**
+ * Internal dependencies
+ */
+import {
+  THEME_CONSTANTS,
+  Text,
+  Toggle,
+  Headline,
+} from '../../../../../../design-system';
+import { FullWidthWrapper } from '../../common/styles';
+import PillGroup from '../../shared/pillGroup';
+import localStore, {
+  LOCAL_STORAGE_PREFIX,
+} from '../../../../../utils/localStore';
+import useLibrary from '../../../useLibrary';
+import useStory from '../../../../../app/story/useStory';
+import {
+  getInUseFontsForPages,
+  getTextSetsForFonts,
+} from '../../../../../utils/getInUseFonts';
+import { Container as SectionContainer } from '../../../common/section';
+import { virtualPaneContainer } from '../../shared/virtualizedPanelGrid';
+import TextSets from './textSets';
+import { CATEGORIES, PANE_TEXT } from './constants';
+
+const TitleBar = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding: 20px 0;
+`;
+
+const TextSetsToggle = styled.div`
+  display: flex;
+  p {
+    margin: auto 12px;
+  }
+`;
+
+const TextSetsWrapper = styled.div`
+  ${virtualPaneContainer};
+`;
+
+function TextSetsPane({ paneRef }) {
+  const { textSets } = useLibrary(({ state: { textSets } }) => ({ textSets }));
+  const [showInUse, setShowInUse] = useState(false);
+
+  const allTextSets = useMemo(() => Object.values(textSets).flat(), [textSets]);
+  const storyPages = useStory(({ state: { pages } }) => pages);
+
+  const [selectedCat, setSelectedCat] = useState(
+    localStore.getItemByKey(`${LOCAL_STORAGE_PREFIX.TEXT_SET_SETTINGS}`)
+      ?.selectedCategory
+  );
+
+  const getTextSetsForInUseFonts = useCallback(
+    () =>
+      getTextSetsForFonts({
+        fonts: getInUseFontsForPages(storyPages),
+        textSets: selectedCat ? textSets[selectedCat] : allTextSets,
+      }),
+    [allTextSets, storyPages, selectedCat, textSets]
+  );
+
+  const addIdToTextSets = useCallback(
+    (sets) =>
+      sets.map((elements) => {
```

Comment:
Probably not a big deal, but it may be confusing to see `sets` here since this is an `Array` and not a `Set`. Maybe we could change the name to something like `textSetsArray`?

Reply:
good call, was so used to set as in textSet didn't realize that would be potentially confusing.